cmd/generate-hashes/generate-hashes.go

```diff
@@ -31,16 +39,36 @@ func main() {
 		os.Exit(1)
 	}
 
+	// set up hasher
+	hashFunc := func() hash.Hash {
+		return sha3.NewLegacyKeccak256()
+	}
+	hashPool := legacy.NewTreePool(hashFunc, 128, legacy.PoolSize)
+	bmtHash := legacy.New(hashPool)
+
 	// create sequence generator and outputs
 	var i int
 	g := mockbytes.New(0, mockbytes.MockTypeStandard).WithModulus(255)
 	for i = 0; i < 4096; i++ {
 		s := fmt.Sprintf("processing %d...", i)
 		fmt.Fprintf(os.Stderr, "%-64s\r", s)
-		filename := fmt.Sprintf(".data/%d.bin", i)
+		filename := fmt.Sprintf("%s/%d.bin", outputDir, i)
 		b, err := g.SequentialBytes(i)
 		if err != nil {
 			fmt.Fprint(os.Stderr, err.Error())
+			os.Exit(1)
+		}
+		bmtHash.Reset()
+		_, err = bmtHash.Write(b)
+		sum := bmtHash.Sum(nil)
+		if err != nil {
+			fmt.Fprint(os.Stderr, err.Error())
+			os.Exit(1)
+		}
+		err = ioutil.WriteFile(filename, sum, 0644)
```

Comment:
My suggestion is to use 0666 file mode, as well as a few lines below, which I probably missed to see in previous pr.

Reply:
:guitar: